Libraries/LibWeb/TrustedTypes/TrustedTypePolicy.cpp

```diff
@@ -169,4 +186,131 @@ WebIDL::ExceptionOr<JS::Value> TrustedTypePolicy::get_trusted_type_policy_value(
+    arguments.append(JS::PrimitiveString::create(vm, to_string(trusted_type_name)));
+    arguments.append(JS::PrimitiveString::create(vm, to_string(sink)));
+    auto policy_value = TRY(default_policy->get_trusted_type_policy_value(
+        trusted_type_name,
+        input.visit(
+            [](auto& value) { return value->to_string(); },
+            [](Utf16String& value) { return value; }),
+        arguments,
+        ThrowIfCallbackMissing::No));
+
+    //  4. If policyValue is null or undefined, return policyValue.
+    if (policy_value.is_nullish())
+        return policy_value;
+
+    //  5. Let dataString be the result of stringifying policyValue.
+    auto data_string = TRY(policy_value.to_utf16_string(vm));
+
+    //  6. Return a new instance of an interface with a type name trustedTypeName, with its associated data value set to dataString.
+    switch (trusted_type_name) {
+    case TrustedTypeName::TrustedHTML:
+        return realm.create<TrustedHTML>(realm, move(data_string));
+    case TrustedTypeName::TrustedScript:
+        return realm.create<TrustedScript>(realm, move(data_string));
+    case TrustedTypeName::TrustedScriptURL:
+        return realm.create<TrustedScriptURL>(realm, move(data_string));
+    }
+    VERIFY_NOT_REACHED();
+}
+
+// https://www.w3.org/TR/trusted-types/#get-trusted-type-compliant-string-algorithm
+WebIDL::ExceptionOr<Utf16String> get_trusted_type_compliant_string(TrustedTypeName expected_type, JS::Object& global, Variant<GC::Root<TrustedHTML>, GC::Root<TrustedScript>, GC::Root<TrustedScriptURL>, Utf16String> input, InjectionSink sink, String sink_group)
+{
+    auto& vm = global.vm();
+
+    // 1. If input is an instance of expectedType, return stringified input and abort these steps.
+    switch (expected_type) {
+    case TrustedTypeName::TrustedHTML:
+        if (auto* const value = input.get_pointer<GC::Root<TrustedHTML>>(); value)
+            return (*value)->to_string();
+        break;
+    case TrustedTypeName::TrustedScript:
+        if (auto* const value = input.get_pointer<GC::Root<TrustedScript>>(); value)
+            return (*value)->to_string();
+        break;
+    case TrustedTypeName::TrustedScriptURL:
+        if (auto* const value = input.get_pointer<GC::Root<TrustedScriptURL>>(); value)
+            return (*value)->to_string();
+        break;
+    }
+
+    // 2. Let requireTrustedTypes be the result of executing Does sink type require trusted types? algorithm, passing global, sinkGroup, and true.
+    auto const require_trusted_types = ContentSecurityPolicy::Directives::does_sink_require_trusted_types(global, sink_group, ContentSecurityPolicy::Directives::IncludeReportOnlyPolicies::Yes);
+
+    // 3. If requireTrustedTypes is false, return stringified input and abort these steps.
+    if (!require_trusted_types)
+        return input.visit([](auto&& value) { return value->to_string(); }, [](Utf16String& value) { return value; });
```

Comment:
suggestion: This can put across multiple lines:
```suggestion
    if (!require_trusted_types) {
        return input.visit(
            [](auto& value) {
                return value->to_string();
            },
            [](Utf16String& value) {
                return value;
            });
    }
```